### ISSUE TITLE
lsp: completion not working reliably inside layouts

### ISSUE THREAD

**Issue by tronical** [slint-ui/slint#2168]
Sometimes the LSP does not offer completion of types at times when it's not obvious.

The following steps reproduce this for me with the VS Code extension:

1. Open a new empty test.slint file.
1. Restart the LSP by choosing "Developer: Reload Window" from the command palette.
2. Type in the following code (don't use copy & paste):
```
export component Demo {
    VerticalLayout {  }
}
```
(observe how "VerticalLayout" is offered for completion when instantiating it)
3. Place the cursor into the `VerticalLayout {}` and try to insert a `Rectangle` or any other type that would normally be completed --> No completion is offered

**Comment by ogoffart** [comment]
I cannot reproduce this.

One thing I can  reproduce if i just completed an element (eg, `VerticalLayout`)
And then ` VerticalLayout{ }` is completed, with the cursor inside the braces.
Then Typing `Rect` doesnt trigger an automatic auto-completion anymore.  But ctrl+space makes it work.
Also changing the cursor position with arrow makes it work.
I think just after a previous completion, vscode won't offer completion immediatly if you're in the field that was just completed.

**Comment by tronical** [comment]
> And then  VerticalLayout{ } is completed, with the cursor inside the braces.

Yes, that is indeed key to reproducing it.

And yes, that doesn't seem to be a bug on our end but VS code / Monaco behavior.
